### PR TITLE
Adding code for Layer_version resource

### PR DESCRIPTION
**Pull request by Vandita2020**
Description of changes:
Adding code for Layer_version resource. This contains hook code and e2e tests for creation and deletion. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_layer_version.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        cr["spec"]["description"] = "new description"
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+        time.sleep(UPDATE_WAIT_AFTER_SECONDS)
+
+        # Check layer version  description
+        layer_version = lambda_validator.get_layer_version(resource_name, version_number)
+        assert layer_version is not None
+        assert layer_version["Description"] == "new description"
```
Comment (A-Hilaly) [inline test/e2e/tests/test_layer_version.py:96]: This is also testing resource updates? Is it planned for this PR?

File: config/controller/kustomization.yaml [hunk 2]
```diff
@@ -6,4 +6,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: public.ecr.aws/aws-controllers-k8s/lambda-controller
-  newTag: v0.1.3
+  newTag: v0.1.2
```
Comment (A-Hilaly) [inline config/controller/kustomization.yaml:9]: This should stay `v0.1.3`, make sure to run `git fetch --tags --all` in lambda-controller repo before regenerating the code

File: apis/v1alpha1/ack-generate-metadata.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 ack_generate_info:
-  build_date: "2022-11-17T22:06:28Z"
-  build_hash: 7cd74c3a855e38ea3fb5e1cf40b26fbe96283870
+  build_date: "2022-12-02T17:17:38Z"
+  build_hash: 560c702f71f3f9041b97417e84076d12721d9629
   go_version: go1.19
-  version: v0.19.3-15-g7cd74c3-dirty
-api_directory_checksum: f01a03847a7bb58e935fdca90ecfb910f645ea61
+  version: v0.19.3-9-g560c702
```
Comment (A-Hilaly) [inline apis/v1alpha1/ack-generate-metadata.yaml:5]: Maybe time to use code-generator v0.20.1? `git fetch --tags --all` and rebasing will help

File: helm/templates/cluster-role-controller.yaml [hunk 4]
```diff
@@ -4,19 +4,11 @@ kind: ClusterRole
...
-  {{- end }}
 {{ else }}
 kind: Role
 metadata:
   creationTimestamp: null
   name: ack-lambda-controller
-  labels:
-  {{- range $key, $value := .Values.role.labels }}
-    {{ $key }}: {{ $value | quote }}
-  {{- end }}
```
Comment (A-Hilaly) [inline helm/templates/cluster-role-controller.yaml:19]: Looks like you're using an outdated code-generator

File: test/e2e/tests/test_layer_version.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        lambda_validator = LambdaValidator(lambda_client)
+
+        # Assert that the original code.s3Bucket and code.s3Key is still part of
+        # the function's CR
+        assert cr["spec"]["content"]["s3Bucket"] == resources.FunctionsBucket.name
+        assert cr["spec"]["content"]["s3Key"] == LAMBDA_FUNCTION_FILE_ZIP
+        version_number = cr['status']['versionNumber']
```
Comment (A-Hilaly) [inline test/e2e/tests/test_layer_version.py:81]: Let's address those in a seperate PR

**Review by A-Hilaly — COMMENTED** [review]
Good start on this @Vandita2020 ! Left few comments below

File: apis/v1alpha1/ack-generate-metadata.yaml (hunk 3 above)
Comment (Vandita2020) [reply apis/v1alpha1/ack-generate-metadata.yaml]: Done.

File: config/controller/kustomization.yaml (hunk 2 above)
Comment (Vandita2020) [reply config/controller/kustomization.yaml]: Done

File: test/e2e/tests/test_layer_version.py (hunk 5 above)
Comment (Vandita2020) [reply test/e2e/tests/test_layer_version.py]: Thanks, I will open separate PR for this.

File: helm/templates/cluster-role-controller.yaml (hunk 4 above)
Comment (Vandita2020) [reply helm/templates/cluster-role-controller.yaml:19]: Done, I have updated the code-generator.

File: test/e2e/tests/test_layer_version.py (hunk 1 above)
Comment (Vandita2020) [reply test/e2e/tests/test_layer_version.py]: Thanks for reminding, this PR is for creation and deletion only. I will make separate PR for update.

File: apis/v1alpha1/ack-generate-metadata.yaml [hunk 6]
```diff
@@ -1,13 +1,13 @@
 ack_generate_info:
-  build_date: "2022-11-17T22:06:28Z"
-  build_hash: 7cd74c3a855e38ea3fb5e1cf40b26fbe96283870
+  build_date: "2022-12-02T19:05:43Z"
+  build_hash: 560c702f71f3f9041b97417e84076d12721d9629
   go_version: go1.19
-  version: v0.19.3-15-g7cd74c3-dirty
-api_directory_checksum: f01a03847a7bb58e935fdca90ecfb910f645ea61
+  version: v0.20.1-2-g560c702
```
Comment (A-Hilaly) [inline apis/v1alpha1/ack-generate-metadata.yaml:5]: 👍

File: helm/templates/cluster-role-controller.yaml (hunk 4 above)
Comment (A-Hilaly) [reply helm/templates/cluster-role-controller.yaml:19]: This code should not be removed. Can you double check that you included changes to `helm/templates` folder?

**Review by A-Hilaly — COMMENTED** [review]
This looks good to go! one last comment bellow.
Good work on this @Vandita2020 !

**Review by A-Hilaly — APPROVED** [review]
Tres bien!
/lgtm

**Pull request merged**